scripts/DP27-JP/c100427002.lua

```diff
@@ -0,0 +1,107 @@
+--D・スキャナン
+--
+--Script by Trishula9
+function c100427002.initial_effect(c)
+	c:EnableReviveLimit()
+	--splimit
+	local e1=Effect.CreateEffect(c)
+	e1:SetType(EFFECT_TYPE_SINGLE)
+	e1:SetProperty(EFFECT_FLAG_CANNOT_DISABLE+EFFECT_FLAG_UNCOPYABLE)
+	e1:SetCode(EFFECT_SPSUMMON_CONDITION)
+	c:RegisterEffect(e1)
+	--special summon
+	local e2=Effect.CreateEffect(c)
+	e2:SetType(EFFECT_TYPE_FIELD)
+	e2:SetCode(EFFECT_SPSUMMON_PROC)
+	e2:SetProperty(EFFECT_FLAG_CANNOT_DISABLE+EFFECT_FLAG_UNCOPYABLE)
+	e2:SetRange(LOCATION_HAND)
+	e2:SetCondition(c100427002.spcon)
+	e2:SetOperation(c100427002.spop)
+	c:RegisterEffect(e2)
+	--search
+	local e3=Effect.CreateEffect(c)
+	e3:SetDescription(aux.Stringid(100427002,0))
+	e3:SetCategory(CATEGORY_TOHAND+CATEGORY_SEARCH+CATEGORY_TODECK)
+	e3:SetType(EFFECT_TYPE_IGNITION)
+	e3:SetRange(LOCATION_MZONE)
+	e3:SetCountLimit(1)
+	e3:SetCondition(c100427002.srcon)
+	e3:SetTarget(c100427002.srtg)
+	e3:SetOperation(c100427002.srop)
+	c:RegisterEffect(e3)
+	--recover
+	local e4=Effect.CreateEffect(c)
+	e4:SetDescription(aux.Stringid(100427002,1))
+	e4:SetCategory(CATEGORY_TOHAND+CATEGORY_TODECK)
+	e4:SetType(EFFECT_TYPE_IGNITION)
+	e4:SetRange(LOCATION_MZONE)
+	e4:SetCountLimit(1)
+	e4:SetCondition(c100427002.rccon)
+	e4:SetTarget(c100427002.rctg)
+	e4:SetOperation(c100427002.rcop)
+	c:RegisterEffect(e4)
+end
+function c100427002.spfilter(c)
+	return c:IsSetCard(0x26) and c:IsType(TYPE_MONSTER) and c:IsAbleToRemoveAsCost()
+end
+function c100427002.spcon(e,c)
+	if c==nil then return true end
+	local tp=c:GetControler()
+	return Duel.GetLocationCount(tp,LOCATION_MZONE)>0
+		and Duel.IsExistingMatchingCard(c100427002.spfilter,tp,LOCATION_HAND,0,1,e:GetHandler())
+end
+function c100427002.spop(e,tp,eg,ep,ev,re,r,rp,c)
+	Duel.Hint(HINT_SELECTMSG,tp,HINTMSG_REMOVE)
+	local g=Duel.SelectMatchingCard(tp,c100427002.spfilter,tp,LOCATION_HAND,0,1,1,e:GetHandler())
+	Duel.Remove(g,POS_FACEUP,REASON_COST)
+end
+function c100427002.srcon(e,tp,eg,ep,ev,re,r,rp)
+	return not e:GetHandler():IsDisabled() and e:GetHandler():IsAttackPos()
+end
+function c100427002.srfilter(c)
+	return c:IsSetCard(0x26) and c:IsType(TYPE_SPELL+TYPE_TRAP) and c:IsAbleToHand()
+end
+function c100427002.srtg(e,tp,eg,ep,ev,re,r,rp,chk)
+	if chk==0 then return Duel.IsExistingMatchingCard(c100427002.srfilter,tp,LOCATION_DECK,0,1,nil) end
+	Duel.SetOperationInfo(0,CATEGORY_TOHAND,nil,1,tp,LOCATION_DECK)
+	Duel.SetOperationInfo(0,CATEGORY_TODECK,nil,1,tp,LOCATION_HAND)
+end
+function c100427002.srop(e,tp,eg,ep,ev,re,r,rp)
+	Duel.Hint(HINT_SELECTMSG,tp,HINTMSG_ATOHAND)
+	local g=Duel.SelectMatchingCard(tp,c100427002.srfilter,tp,LOCATION_DECK,0,1,1,nil)
+	if g:GetCount()>0 and Duel.SendtoHand(g,nil,REASON_EFFECT)>0 then
```

Comment:
According to the current text on Yugipedia, the card added using Scannen's effects must land in the hand before a card will be returned to the top of the Deck.
```suggestion
	if g:GetCount()>0 and Duel.SendtoHand(g,nil,REASON_EFFECT)>0 and g:GetFirst():IsLocation(LOCATION_HAND) then
```